internal/alloycli/cmd_run.go

```diff
@@ -161,9 +163,16 @@ type alloyRun struct {
 	configFormat                 string
 	configBypassConversionErrors bool
 	configExtraArgs              string
+	enableAutoMemoryLimit        bool
 }
 
 func (fr *alloyRun) Run(configPath string) error {
+	// Set the memory limit, this will honor GOMEMLIMIT if set
+	// If there is a cgroup will follow that
+	if fr.enableAutoMemoryLimit {
+		memlimit.SetGoMemLimitWithOpts()
```

Comment:
Is it possible to pass in a compatible logger here? If so, some of the logs might be helpful (e.g. if `GOMEMLIMIT` is already set it won't honor the cgroups settings, as well as confirmation of the limit)

Reply:
Was thinking something like from the example:
```golang
memlimit.SetGoMemLimitWithOpts(
		memlimit.WithRatio(0.9),
		memlimit.WithProvider(memlimit.FromCgroup),
		memlimit.WithLogger(slog.Default()),          // Can we pass a logger here?
	)
```

Reply:
I can but I have to move it til after our logging is fully initialized, which likely isn't the end of the world its just later in the process to pass down the bool  flag.

Reply:
Added the slog.Default, its...a bit different looking but likely works for ease of use.

`2024/04/23 17:01:27 INFO memory is not limited, skipping: %v package=github.com/KimMachineGun/automemlimit/memlimit !BADKEY="memory is not limited"`